psi4/driver/p4util/python_helpers.py

```diff
@@ -75,7 +75,16 @@ def pybuild_basis(mol, key=None, target=None, fitrole='ORBITAL', other=None, pur
         core.print_out(basisdict['message'])
 
     psibasis = core.BasisSet.construct_from_pydict(mol, basisdict, puream)
-    return psibasis
+    ecpbasis = None
+    if 'ecp_shell_map' in basisdict:
+        ecpbasis = core.BasisSet.construct_ecp_from_pydict(mol, basisdict, puream)
+
+    if key == 'BASIS':
+        # For orbitals basis sets, we need to return ECP also
+        return psibasis, ecpbasis
```

Comment:
This will break a great deal of code. Is there any way we can simplify this such as adding the ECP basis object to the canonical Basis or only returning the ECP part if requested?